hivemind/dht/dht.py

```diff
@@ -72,7 +72,7 @@ def __init__(
         self.num_workers = num_workers
 
         self._record_validator = CompositeValidator(record_validators)
-        self._inner_pipe, self._outer_pipe = mp.Pipe(duplex=True)
+        self._inner_pipe, self._outer_pipe = mp.Pipe(duplex=False)
```

Comment:
Everything works with duplex=False according to tests, best to use pipes since duplex relies on socketpair, which is slower [[1]](https://stackoverflow.com/a/3518624) [[2]](https://stackoverflow.com/a/54164058)